src/Cli/dotnet/commands/dotnet-workload/list/VisualStudioWorkloads.cs

```diff
@@ -0,0 +1,172 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using Microsoft.Deployment.DotNet.Releases;
+using Microsoft.DotNet.Workloads.Workload.Install.InstallRecord;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using Microsoft.VisualStudio.Setup.Configuration;
+
+#nullable disable
+
+namespace Microsoft.DotNet.Workloads.Workload.List
+{
+    /// <summary>
+    /// Provides functionality to query the status of .NET workloads in Visual Studio.    
+    /// </summary>
+#if NETCOREAPP
+    [SupportedOSPlatform("windows")]
+#endif
+    internal static class VisualStudioWorkloads
+    {
+        private const int REGDB_E_CLASSNOTREG = unchecked((int)0x80040154);
+
+        /// <summary>
+        /// Visual Studio product ID filters. We dont' want to query SKUs such as Server, TeamExplorer, TestAgent
+        /// TestController and BuildTools.
+        /// </summary>
+        private static readonly string[] s_visualStudioProducts = new string[]
+        {
+            "Microsoft.VisualStudio.Product.Community",
+            "Microsoft.VisualStudio.Product.Professional",
+            "Microsoft.VisualStudio.Product.Enterprise",
+        };
+
+        /// <summary>
+        /// The SWIX package ID wrapping the SDK installer in Visual Studio. The ID should contain
+        /// the SDK version as a suffix, e.g., "Microsoft.NetCore.Toolset.5.0.403".
+        /// </summary>
+        private static readonly string s_visualStudioSdkPackageIdPrefix = "Microsoft.NetCore.Toolset.";
+
+        /// <summary>
+        /// Gets a set of workload components based on the available set of workloads for the current SDK.
+        /// </summary>
+        /// <param name="workloadResolver">The workload resolver used to obtain available workloads.</param>
+        /// <returns>A collection of Visual Studio component IDs corresponding to workload IDs.</returns>
+        internal static IEnumerable<string> GetAvailableVisualStudioWorkloads(IWorkloadResolver workloadResolver) =>
+            workloadResolver.GetAvailableWorkloads().Select(w => w.Id.ToString().Replace('-', '.'));
+
+        /// <summary>
+        /// Finds all workloads installed by all Visual Studio instances given that the
+        /// SDK installed by an instance matches the feature band of the currently executing SDK.
+        /// </summary>
+        /// <param name="workloadResolver">The workload resolver used to obtain available workloads.</param>
+        /// <param name="sdkFeatureBand">The feature band of the executing SDK.</param>
+        /// <param name="installedWorkloads">The collection of installed workloads to update.</param>
+        internal static void GetInstalledWorkloads(IWorkloadResolver workloadResolver, SdkFeatureBand sdkFeatureBand,
+            ref InstalledWorkloadsCollection installedWorkloads)
```

Comment:
As far as I can see, there's no need to pass in `installedWorkloads` via `ref`.  The method modifies the collection but doesn't change the reference passed in.